dongle/mt76.cpp

```diff
@@ -1408,6 +1408,37 @@ bool Mt76::initChannels()
     return true;
 }
 
+bool Mt76::openFirmware()
+{
+    const std::string filename(DATADIR "/FW_ACC_00U.bin");
+
+    if (firmware.size() != 0)
+    {
+        Log::debug("Firmware binary already loaded.");
+        return true;
+    }
+
+    std::ifstream blob(filename, std::ios::binary);
+    if (!blob)
+    {
+        Log::error("Failed to open firmware binary %s\n\n"
+	"You can get the firmware using the following commands.\n\n"
```

Comment:
I wonder if these instructions should be moved to a man page instead, referencing just the man page here. In turn, we could add an additional check `ConditionFileNotEmpty=` in the systemd service file. The service could then reference the same man page (`Documentation=` in the systemd unit).

Otherwise we would end up with a lot of log spam each time the service starts (and tries to restart due to failure for multiple times).

Reply:
Fully agree on all points - although there is no man page at the moment to update :frowning: 
Don't mind writing something as a start, but would love to hear if the maintainer is happy with the approach.

Reply:
We could simply link to an installed markdown text file installed with the package instead.

Reply:
For the Feral gamemoded logger I wrote a macro that could log items with a hint once, and if logging the same entry again, the hint text would be omitted. It also prefixed each hint line with `  --` to more easily see or filter those entries.

Reply:
Cannot see a random markdown file bringing any benefits over a man page. Especially since one could convert markdown to man with `scdoc`. Will try to polish it up this week.

Reply:
In light of https://github.com/medusalix/xow/commit/700529b2517df7d17ad5a2fa0bd679143ac48666 I will be keeping the MR open, but won't update unless a maintainer weights-in.

Reply:
Fwiw I moved the extraction magic to a standalone script, which is referenced by `xow`.